docs/development/development.md

```diff
@@ -0,0 +1,258 @@
+
+</details>
+
+---
+
+### **Are there any coding standards in place? For example, do we use structs with methods instead of functions?**
+
+<details>
+<summary>Answer</summary>
+
+The project follows the typical Go coding standards. Specifically, structs with associated methods are preferred over using standalone functions. This helps encapsulate business logic better and makes the code more organized and maintainable. Additionally, Go’s naming and formatting conventions are followed, such as using lowercase letters for private variables and methods.
+
+</details>
+
+---
+
+### **How to set up a local development environment, using the Makefile?**
+
+<details>
+<summary>Answer</summary>
+
+To set up a local development environment, it is recommended to use an editor that supports Go debugging, such as Visual Studio Code or GoLand. The Makefile includes several useful commands for the development environment:
+
+- `make tools`: Installs the necessary dependencies for the project.
+- `make format-go`: Runs Go formatting, useful if you don't have automatic formatting enabled in the editor.
+- `make license-add`: Adds a license to newly added files.
+- `make lint-go`: Runs Go linting to check for style issues or common errors.
+- `make test-unit`: Runs unit tests.
+- `make test-integration`: Runs integration tests.
+
+Other commands in the Makefile are less relevant during the development cycle, so these are the main ones to use.
+
+</details>
+
+---
+
+### **What's the release process for a new version?**
+
+<details>
+<summary>Answer</summary>
+
+The release process for a new version is documented at [this link](https://github.com/sighupio/fury-distribution/blob/main/MAINTENANCE.md#furyctl). If the release is not tied to `fury-distribution`, it's enough to create a tag and release it. However, if the release is dependent on new versions of the distribution, the process may be more complex and require updating `fury-distribution` before releasing a new version.
+
+</details>
+
+---
+
+### **How does the template engine works and what are the available features?**
+
+<details>
+<summary>Answer</summary>
+
+The template engine used is the standard Go template engine, which also leverages the `https://github.com/Masterminds/sprig` library. Sprig provides several additional functions for templates, such as string manipulations, date formatting, and other utilities not included in Go's native template engine.
+
+</details>
+
+---
+
+### **How logging is implemented, libraries used, and conventions (e.g., log messages)?**
+
+<details>
+<summary>Answer</summary>
+
+Logging in the project is implemented using the `https://github.com/sirupsen/logrus` library, which is one of the most popular logging libraries for Go. There are no strict conventions for log messages, but Logrus supports various logging levels (info, error, debug) and can output logs in different formats, including plain text and JSON.
```

Comment:
Let's add some informations about the furyctl.log logic and where it's located in the code

Reply:
There are at least the following conventions regarding logs:

1. messages that the user sees by default should provide useful information and not leak implementation details. For example:

BAD:
`INFO Running ansible playbooks`

GOOD:
`INFO Installing Kubernetes packages in the nodes`

2. (I'm 99% sure about this): all the tools that we call we configure them to output structured logs and are wrapped in furyctl structured logs in the log file.

3. A log file is created for each time you run furyctl, the log file has a timestamp and some other random int in the file name. I don't know the reasoning behind this.